sonar-web-plugin/src/main/java/org/sonar/plugins/web/rules/WebRulesDefinition.java

```diff
@@ -17,98 +17,44 @@
  */
 package org.sonar.plugins.web.rules;
 
-import com.google.common.base.Charsets;
-import com.google.common.base.Preconditions;
-import com.google.common.io.Resources;
-import com.google.gson.Gson;
-import org.sonar.api.rule.RuleStatus;
-import org.sonar.api.rules.RuleType;
-import org.sonar.api.server.debt.DebtRemediationFunction;
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 import org.sonar.api.server.rule.RulesDefinition;
 import org.sonar.plugins.web.api.WebConstants;
-import org.sonar.squidbridge.annotations.AnnotationBasedRulesDefinition;
-
-import javax.annotation.Nullable;
-import java.io.IOException;
-import java.net.URL;
-import java.util.Locale;
+import org.sonarsource.analyzer.commons.RuleMetadataLoader;
 
 public final class WebRulesDefinition implements RulesDefinition {
 
   public static final String REPOSITORY_KEY = WebConstants.LANGUAGE_NAME;
   public static final String REPOSITORY_NAME = "SonarAnalyzer";
 
-  private final Gson gson = new Gson();
+  private static final Set<String> TEMPLATE_RULE_KEYS = Collections.unmodifiableSet(Stream.of(
+    "IllegalAttributeCheck",
+    "LibraryDependencyCheck",
+    "ChildElementIllegalCheck",
+    "ChildElementRequiredCheck",
+    "ParentElementIllegalCheck",
+    "ParentElementRequiredCheck").collect(Collectors.toSet()));
 
   @Override
   public void define(Context context) {
     NewRepository repository = context
       .createRepository(REPOSITORY_KEY, WebConstants.LANGUAGE_KEY)
       .setName(REPOSITORY_NAME);
 
-    new AnnotationBasedRulesDefinition(repository, WebConstants.LANGUAGE_KEY).addRuleClasses(false, CheckClasses.getCheckClasses());
-    for (NewRule rule : repository.rules()) {
-      String metadataKey = rule.key();
-      rule.setInternalKey(metadataKey);
-      addMetadata(rule, metadataKey);
-    }
-    repository.done();
-  }
+    // FIXME: with SonarQube 6.7, should use the sonar way profile location as extra parameter
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Take the required action to fix the issue indicated by this comment. ([squid:S1134](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.web%3Aweb&branch=SONARWEB-66&issues=AWGeex3xjNA6aRBI4SDV&open=AWGeex3xjNA6aRBI4SDV)

[ISSUE_KEY]: AWGeex3xjNA6aRBI4SDV